pkg/common/jwtsvid/simple.go

```diff
@@ -0,0 +1,162 @@
+package jwtsvid
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"time"
+
+	jwt "github.com/dgrijalva/jwt-go"
+	"github.com/spiffe/spire/pkg/common/cryptoutil"
+	"github.com/spiffe/spire/pkg/common/idutil"
+)
+
+const (
+	certificateThumbprintHeader = "x5t#S256"
+)
+
+func SignSimpleToken(spiffeID string, audience []string, expires time.Time, signer crypto.Signer, cert *x509.Certificate) (string, error) {
+	if err := validateKeypair(signer, cert); err != nil {
+		return "", err
+	}
+
+	if err := idutil.ValidateSpiffeID(spiffeID, idutil.AllowAnyTrustDomainWorkload()); err != nil {
+		return "", err
+	}
+
+	// cap expiration to the signing certificate expiration, if necessary.
+	if expires.IsZero() {
+		return "", errors.New("expiration is required")
+	}
+	if !cert.NotAfter.IsZero() && expires.After(cert.NotAfter) {
```

Comment:
I don't think it will ever be zero in practice. This code was intended to be somewhat generic so I was covering my bases. That is also why this code caps the expiration to the certificate as well as the ServerCA (although I'm ok removing one or the other). I'm going to remove this check here, in favor of the one in the server CA.